flink-cdc-common/src/main/java/org/apache/flink/cdc/common/utils/SchemaUtils.java

```diff
@@ -452,6 +455,43 @@ public static Schema inferWiderSchema(@Nullable Schema lSchema, Schema rSchema)
         return lSchema.copy(mergedColumns);
     }
 
+    public static void validateMetaSchemaCompatibility(LinkedHashSet<Schema> schemas) {
+        if (schemas.size() > 1) {
+            Schema outputSchema = null;
+            for (Schema schema : schemas) {
+                isMetaSchemaCompatible(outputSchema, schema);
+                outputSchema = schema;
+            }
+        }
+    }
+
+    public static void isMetaSchemaCompatible(
```

Comment:
nit: this is not a pure function returning `Boolean`, so `validateSchemaMetadataCompatibility` might be a better name.